### PR TITLE
Add jaeger package

### PR DESCRIPTION
**Pull request by saulshanabrook**
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x]  If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by saulshanabrook** [comment]
@conda-forge-linter Thank you!

**Comment by saulshanabrook** [comment]
I see it's giving this error:

```
patchelf: cannot find section '.dynamic'. The input file is most likely statically linked
patchelf: --print-rpath failed for /home/conda/staged-recipes/build_artifacts/jaeger_1571776179014/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac/bin/jaeger-collector
```

Is that just a warning? It doesn't seem to be finding the binaries.

File: recipes/jaeger/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: {{name }}-{{ version }}-darwin-amd64.tar.gz  # [osx]
+  url: https://github.com/jaegertracing/{{ name }}/releases/download/v{{ version }}/{{name }}-{{ version }}-darwin-amd64.tar.gz  # [osx]
+  sha256: 06046f509b6a72de2144a12426be835f5c3903939a6337a70b6dc524cac7bd47  # [osx]
+
+  fn: {{name }}-{{ version }}-linux-amd64.tar.gz  # [linux64]
```
Comment (scopatz) [inline recipes/jaeger/meta.yaml]: The selector is just `[linux]`

File: recipes/jaeger/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  url: https://github.com/jaegertracing/{{ name }}/releases/download/v{{ version }}/{{name }}-{{ version }}-linux-amd64.tar.gz  # [linux64]
+  sha256: aa1764b8cb5a0cbdb128f3403e25b1fad240734da20460ec13ac89674abe3d8b  # [linux64]
+
+  fn: {{name }}-{{ version }}-windows-amd64.tar.gz  # [win64]
```
Comment (scopatz) [inline recipes/jaeger/meta.yaml]: The selector is just `[win]`

File: recipes/jaeger/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
+{% set name = "jaeger" %}
+{% set version = "1.14.0" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: {{name }}-{{ version }}-darwin-amd64.tar.gz  # [osx]
```
Comment (scopatz) [inline recipes/jaeger/meta.yaml]: Is there a reason that this cannot be compiled from source?  We generally do not repackage binaries for compatability reasons

Comment (saulshanabrook) [reply recipes/jaeger/meta.yaml]: No reason besides laziness.

It's all statically linked, I am curious what the compatibility issues would be.

I copied the structure from this go package which also just downloads the binaries: https://github.com/conda-forge/go-spatial-feedstock/blob/master/recipe/meta.yaml

Are there best practices for distributing go binaries in conda forge?

**Comment by saulshanabrook** [comment]
Thanks @scopatz for looking through this.

File: recipes/jaeger/meta.yaml (hunk 3 above)
Comment (scopatz) [reply recipes/jaeger/meta.yaml]: Oh if this is a go library, I am not sure what the best practices are.  CC @conda-forge/help-go!

**Comment by scopatz** [comment]
Also CC @conda-forge/staged-recipes

File: recipes/jaeger/meta.yaml (hunk 3 above)
Comment (mariusvniekerk) [reply recipes/jaeger/meta.yaml]: So for golang stuff we tend to just build it.  See our terraform / terragrunt feedstocks

Comment (saulshanabrook) [reply recipes/jaeger/meta.yaml]: One of the advantages of building with Go is that it's really easy to cross compile binaries. Is there a way in conda forge to say "We just need to run this one CI job on linux that will produce binaries for linux, mac, and windows"

**Comment by scopatz** [comment]
Seems like it can't find some files

File: recipes/jaeger/meta.yaml (hunk 3 above)
Comment (dharhas) [reply recipes/jaeger/meta.yaml]: So my go-spatial feedstock was created when Go was not available as a conda package so building it here was not an option back then.

File: recipes/jaeger/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  # We must use git here beause the program looks
+  # for the git sha&tag to determine its version
+  - folder: {{ pkg_src }}
+    url: https://{{ goname }}/archive/v{{version}}.tar.gz
+    sha256: 65c39e068cfb23f67529b75617202adca5ab41668942ae91fee14056b7657620 
+
+
+build:
+  number: 0
+  script:
```
Comment (scopatz) [inline recipes/jaeger/meta.yaml]: This should be in separate `build.sh` and `bld.bat` scripts

File: recipes/jaeger/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    - "cp -v examples/hotrod/hotrod-* ${PREFIX}/bin/hotrod"
+    - "chmod -v 755 ${PREFIX}/bin/*"
+
+
+requirements:
+  build:
+    - {{ compiler('go') }}
+    - go-dep
+    - nodejs
+    - yarn
```
Comment (scopatz) [inline recipes/jaeger/meta.yaml]: yarn, node and go seem like host deps to me

Comment (saulshanabrook) [reply recipes/jaeger/meta.yaml]: TBH not sure what the difference is between host deps and these. 

You don't need them once you build the binary, you just need it to build it.

Comment (scopatz) [reply recipes/jaeger/meta.yaml]: build is the OS that the build occurs on (where the compilers are). Host is the OS target (where binaries will be deployed to). This is an especially important distinction for cross-compiling.

**Comment by goanpeca** [comment]
>Seems like it can't find some files

@scopatz question:

The makefile for this recipe depends on git 

```
GIT_SHA=$(shell git rev-parse HEAD)
GIT_CLOSEST_TAG=$(shell git describe --abbrev=0 --tags)
```

Plus we also depend on submodules, so using multiple sources solves 1 issue but does not solve how the package is [built by default](https://github.com/jaegertracing/jaeger/blob/master/Makefile)

Could I use `git_rev` and `git_url` for this case or do I need to patch the makefile to deal with all the places where a git call is used?

Thanks!

**Comment by goanpeca** [comment]
This one should be ready now @conda-forge/staged-recipes.

**Comment by scopatz** [comment]
thanks!

**Pull request merged**